ignite/pkg/buf/buf.go

```diff
@@ -0,0 +1,177 @@
+package buf
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/exec"
+	"github.com/ignite/cli/ignite/pkg/protoanalysis"
+	"github.com/ignite/cli/ignite/pkg/xexec"
+	"github.com/ignite/cli/ignite/pkg/xos"
+)
+
+type (
+	// Command represents a high level command under buf.
+	Command string
+
+	// Buf represents the buf application structure.
+	Buf struct {
+		path     string
+		sdkCache string
+		cache    *protoanalysis.Cache
+	}
+)
+
+const (
+	cosmosSDKModulePath = "github.com/cosmos/cosmos-sdk"
+	binaryName          = "buf"
+	flagTemplate        = "template"
+	flagOutput          = "output"
+	flagErrorFormat     = "error-format"
+	flagLogFormat       = "log-format"
+	fmtJSON             = "json"
+
+	// CMDGenerate generate command.
+	CMDGenerate Command = "generate"
+)
+
+var (
+	commands = map[Command]struct{}{
+		CMDGenerate: {},
+	}
+
+	// ErrInvalidCommand error invalid command name.
+	ErrInvalidCommand = errors.New("invalid command name")
+)
+
+// New creates a new Buf based on the installed binary.
+func New() (Buf, error) {
+	path, err := xexec.ResolveAbsPath(binaryName)
+	if err != nil {
+		return Buf{}, err
+	}
+	return Buf{
+		path:  path,
+		cache: protoanalysis.NewCache(),
+	}, nil
+}
+
+// String returns the command name.
+func (c Command) String() string {
+	return string(c)
+}
+
+// Generate runs the buf Generate command for each file into the proto directory.
+func (b Buf) Generate(ctx context.Context, protoDir, output, template string) (err error) {
+	flags := map[string]string{
+		flagTemplate:    template,
+		flagOutput:      output,
+		flagErrorFormat: fmtJSON,
+		flagLogFormat:   fmtJSON,
+	}
+
+	// TODO find a better way to generate the cosmos-sdk files
+	// the buf.work.yaml contains the `orm/internal` folder, but the `go mod`
+	// can't download this folder because is unused as a dependency. We need to
+	// change the workspace copying the files to another folder and generate the
+	// files.
+	if strings.Contains(protoDir, cosmosSDKModulePath) {
```

Comment:
```suggestion
```

Reply:
@ilgooz, I made a double-check here. We can't remove this validation. Buf will solve the dependencies for the proto files. But we still need to go one by one of the Golang project dependencies and compile the proto files for the ts-client and open API spec generation. This workaround solves the issue regarding the invalid proto path (`orm/internal`) download into the `go mod` for cosmos-sdk buf workspace